### PR TITLE
Restructure reloading and modules in Metricbeat

### PR DESCRIPTION
**Pull request by ruflin**
* Move modules and all related things to its own package including reloading
* Move LoadConfig and GlobWatcher to libbeat for reuse
* Do renaming according to packages

No logic changes were done in this PR.

### CODE REVIEW

**Comment by ruflin** [comment]
@andrewkroh Perhaps I should move reload into the module package as now that I extracted more things to libbeat, there is not too much in it anymore.

File: libbeat/cfgfile/cfgfile.go [hunk 1]
```diff
@@ -138,6 +139,23 @@ func Load(path string) (*common.Config, error) {
 	return config, nil
 }
 
+// LoadConfigs loads a list of configs data from the given file
```
Comment (andrewkroh) [inline libbeat/cfgfile/cfgfile.go]: `s/LoadConfigs/LoadList/` and please append a dot to the line.

File: libbeat/cfgfile/cfgfile.go [hunk 2]
```diff
@@ -138,6 +139,23 @@ func Load(path string) (*common.Config, error) {
...
+func LoadList(file string) ([]*common.Config, error) {
+	logp.Debug("cfgfile", "Load config from file: %s", file)
+	rawConfig, err := common.LoadFile(file)
+	if err != nil {
+		return nil, fmt.Errorf("Invalid config: %s", err)
```
Comment (andrewkroh) [inline libbeat/cfgfile/cfgfile.go]: Should we not capitalize the error strings? (https://github.com/golang/go/wiki/CodeReviewComments#error-strings)

File: metricbeat/mb/module/module.go [hunk 3]
```diff
@@ -24,16 +24,16 @@ const (
...
 )
 
-// ModuleWrapper contains the Module and the private data associated with
+// Wrapper contains the Module and the private data associated with
 // running the Module and its MetricSets.
 //
-// Use NewModuleWrapper or NewModuleWrappers to construct new ModuleWrappers.
-type ModuleWrapper struct {
+// Use NewWrapper or NewWrappers to construct new Wrappers.
+type Wrapper struct {
```
Comment (andrewkroh) [inline metricbeat/mb/module/module.go]: WDYT about renaming this file from `module.go` to `wrapper.go` and doing the same with `module_test.go`?

Comment (ruflin) [inline metricbeat/mb/module/module.go]: Done

**Pull request merged**